src/datachain/query/dataset.py

```diff
@@ -519,6 +519,7 @@ def populate_udf_table(self, udf_table: "Table", query: Select) -> None:
                     udf = self.udf
 
                 warehouse = self.catalog.warehouse
+                udf_fields = [str(c.name) for c in query.selected_columns]
```

Comment:
Q: can you please explain / copy paste from the description the relevant part - what is the deal with these `udf_fields`. I'm a bit concerned with code like:

`def _create_worker(self, udf_fields: Sequence[str])`

seems weird that worker needs this (and only this)

Reply:
Before we do pass dicts to UDF as input data. Now we are passing tuples with data only, but we still need to know names of the fields, so we also passing `udf_fields` as list of names of the files (dicts keys in previous implementation). That's why we need it here. I can work a bit more on this because I see this may be improved a bit.